scrapy/http/request/form.py

```diff
@@ -54,10 +55,15 @@ def _urlencode(seq, enc):
     return urlencode(values, doseq=1)
 
 
+def _create_parser_from_response(response, parser_cls):
```

Comment:
> Sorry, I feel dumb - could you please explain what is extended-html and why is it different from html?

FormRequest uses `lxml.html.HTMLParser` which is different from `type=html` in `Selector` class which uses `lxml.etree.HTMLParser`. 

The former provides extra methods to query html in a python way but it is slower than the later.